web/pandas/pdeps/0008-inplace-methods-in-pandas.md

```diff
@@ -0,0 +1,433 @@
+# PDEP-8: In-place methods in pandas
+
+- Created: 16 February 2023
+- Status: Accepted
+- Discussion: [PR 51466](https://github.com/pandas-dev/pandas/pull/51466)
+- Authors: [Thomas Li](https://github.com/lithomas1),
+           [Patrick Hoefler](https://github.com/phofl),
+           [Joris Van den Bossche](https://github.com/jorisvandenbossche)
+- Revision: 1
+
+## Abstract
+
+This PDEP proposes that:
+
+- The ``inplace`` parameter will be removed from any method which can never update the
+  underlying values of a pandas object inplace or which alters the shape of the object,
+  and where the `inplace=True` option is only syntactic sugar for reassigning the result
+  to the calling DataFrame/Series.
+- As a consequence, the `inplace` parameter is only kept for those methods that can
+  modify the underlying values of a pandas object inplace, such as `fillna` or `replace`.
+- With the introduction of Copy-on-Write ([PDEP-7](^1)), users don't need the `inplace`
+  keyword to avoid a copy of the data.
+- For those methods that will keep the `inplace=True` option:
+    - the method will do an attempt to do the operation inplace but still silently copy
+      when needed (for Copy-on-Write), i.e. there is no guarantee it is actually done inplace.
+    - the method will return the calling object (`self`), instead of the current `None`.
```

Comment:
Does this also apply to methods that only act inplace and so have no inplace argument? `update` and `insert` come to mind.

cc @pandas-dev/pandas-core

Reply:
We second this question here https://github.com/pandas-dev/pandas/issues/59788

Reply:
> Does this also apply to methods that only act inplace and so have no inplace argument? `update` and `insert` come to mind.

See related discussion at https://github.com/pandas-dev/pandas/issues/21858